### DISCUSSION TITLE
Model map

### DISCUSSION THREAD

**Discussion by PhiloNL** [livewire/livewire#4074] [3. Feature ideas]
Similar to Laravel's `MorphMap` I would like to suggest the same for Livewire but in this case to expose less internal data. In this example, the  ` \App\Models\User::class` is now referenced as `user`.

I've done some tinkering and got it to work:
https://github.com/livewire/livewire/compare/master...PhiloNL:model-map?expand=1

```php
Livewire::modelMap([
    'user' => App\Models\User::class,
]);
```

```json
{
   "effects":{
   },
   "serverMemo":{
      "dataMeta":{
         "models":{
            "user":{
               "class":"user",
               "id":"95c2bb48-46d5-4881-9389-fd774ffffee5",
               "relations":[
                  "team"
               ],
               "connection":"mysql"
            }
         }
      },
      "checksum":"c2f7a1fb81dd8bb4773427b4e0567e497bf164e2cafa398f31bf18756f6f2fb0"
   }
}
```

I'm curious if we can remove the `connection` from the dataMeta via a config option to let Livewire know the connection is always `mysql`, for example.

**Comment by joshhanley** [comment]
@PhiloNL nice! I think that could be a good feature!

I'm curious though, what is the problem with having the class name and connection name in the metadata? Can you see any issues with exposing that other than it'd be nice not to display the internals?

**Reply by PhiloNL** [reply]
Thanks for your reply @joshhanley

It's mainly from a security perspective; I always prefer giving away as little as possible about the application's inner workings.

For example, the metadata could expose information about a third-party model part of a package with a known exploit. This is useful information to find weaknesses.

I don't have any concerns about leaving it as is for now. But it could be of help to tighten security.

**Reply by joshhanley** [reply]
@PhiloNL yep, fair call. Make sense, that was what I was thinking too, but it's good to confirm 🙂

There is a feature that is being worked on in discussion #3812, that if merged, would allow us to easily add support for things like this in, or users could then also override this functionality themselves or create a package for it.

Check it out, would great to know what you think about that!
